griptape/drivers/web_scraper/markdown_web_scraper_driver.py

```diff
@@ -0,0 +1,61 @@
+from typing import Optional
+from attr import define, field, Factory
+from griptape.drivers import BaseWebScraperDriver
+from griptape.utils import import_optional_dependency
+
+
+@define
+class MarkdownWebScraperDriver(BaseWebScraperDriver):
```

Comment:
Not sure how I feel about the name `MarkdownWebScraperDriver`. The Driver names usually imply the tools they use, not their output format. Is there a reason we're returning in markdown for this Driver but not others?

Reply:
About the naming:

The `TrafilaturaWebScraperDriver` only relies on a single dependency (`Trafilatura`, obviously) to fetch the web page and to extract the meaningful content from it, so here the naming is pretty straight forward, but with this new web scraper, it's using three different dependencies, `playwright` to fetch the web page, `beautifulsoup4` to remove to filter out some of the html, and `markdownify` to convert the html to markdown. Using a name to match the functionality is a little awkward here, since it should effectively be called `PlaywrightBeautifulSoupMarkdownifyWebScraperDriver`, which is horrible.

---

About the output formatting:

The `TrafilaturaWebScraperDriver` currently does apply markdown formatting for links. The class has an `include_links` parameter, which if true (the default) links are converted into markdown (e.g. `<a href="foo.com">foo</a>` to `[foo](foo.com`).

Replicating this behavior with beautiful soup alone is rather low level and doesn't seem worth the trouble. That said,  after seeing the markdown outputted by chat gpt when I told it to 'Extract the meaningful content from this html`, it seems like retaining some minimal document structure would be a good trade off between a conciseness and accuracy (plus its much more human readable if you are debugging). In other words, it seems like the purpose of our web scrapers is to extract the meaningful content from the webpage, and I think markdown is a good choice for that. Let me know if you think otherwise

Reply:
Renamed to `MarkdownifyWebScraperDriver` in order follow the convention without being too verbose.